src/serializers/fields.rs

```diff
@@ -156,9 +156,12 @@ impl GeneralFieldsSerializer {
         let output_dict = PyDict::new(py);
         let mut used_req_fields: usize = 0;
 
-        // NOTE! we maintain the order of the input dict assuming that's right
-        for result in main_iter {
-            let (key, value) = result?;
+        let mut items = main_iter.collect::<PyResult<Vec<_>>>()?;
```

Comment:
My intuition is that this extra Vec being created is the cause for the slowdown in benchmarks.

Reply:
I refactored the function out, so we can re-use this. and avoid creating new variable when sort keys is false